engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/types/TypesInformationExtractor.scala

```diff
@@ -1,18 +1,21 @@
 package pl.touk.nussknacker.engine.types
 
+import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.lang3.ClassUtils
 import pl.touk.nussknacker.engine.api.process.ClassExtractionSettings
 import pl.touk.nussknacker.engine.api.typed.ClazzRef
-import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedDict, TypedObjectTypingResult, TypedTaggedValue, TypedUnion, TypingResult, Unknown}
-import pl.touk.nussknacker.engine.definition.TypeInfos.{ClazzDefinition, MethodInfo}
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.definition.TypeInfos.{ClazzDefinition, MethodInfo, Parameter}
 import pl.touk.nussknacker.engine.types.EspTypeUtils.clazzDefinition
+import pl.touk.nussknacker.engine.util.logging.ExecutionTimeMeasuring
 import pl.touk.nussknacker.engine.variables.MetaVariables
 
-object TypesInformationExtractor {
+import scala.collection.mutable
+
+object TypesInformationExtractor extends LazyLogging with ExecutionTimeMeasuring {
```

Comment:
Let's add at least TODO to document how it really works? I can write sth, but I think that in this case tests are not necessarily enough...

Reply:
You mean in javadocs in code or there: https://nussknacker.io/API.html ? Test case are mainly testing discovery and default configuration. Do you think that we should describe default config or just write how to define own predicates?